### DISCUSSION TITLE
Cli authentication through generated custom key by current_active_user

### DISCUSSION THREAD

**Discussion by nik0811** [fastapi-users/fastapi-users#1137] [Q&A]
After login I am giving an user options to generate api_key which will allow the cli tools or lets assume curl to interact with my api's, Now the issue is where ever I am using `dependencies=[Depends(current_active_user)]` in route the same route should be authenticated by api_key too which am allowing user's to pass in header and that api_key will act as a secret authentication key, Is there any way we can achieve in fastapi-users ?

Below are the code am using for generating the key, but now I want the same key to used for login through cli, 
```
@app.get(f"/{settings.API_HEADER_VERSION}/api/token/generate", dependencies=[Depends(current_active_user)])
async def generate_api_token(current_user: User = Depends(current_active_user)):
    try:
        uuId = str(uuid.uuid4()).replace("-","")
        bytes = uuId.encode('utf-8')
        salt = bcrypt.gensalt()
        hash = bcrypt.hashpw(bytes, salt)
        current_user.api_token=hash
        current_user.api_token_created=datetime.now(timezone("Asia/Kolkata")).strftime('%Y-%m-%d %H:%M:%S')
        update_hash_details = await database_connection()['User'].update_one({"_id": current_user.id}, {"$set": dict(current_user)})
        if update_hash_details.modified_count == 1:
            if (updated_hash_details := await database_connection()['User'].find_one({"_id": current_user.id}) is not None):
                return ResponseModel({"token": uuId, "updated_hash_details": updated_hash_details}, "token generated successfully")
    except:
       return ErrorResponseModel("An error occurred.", 502, "Bad Request.")
```

**Comment by frankie567** [comment]
You need to implement a custom [strategy](https://fastapi-users.github.io/fastapi-users/10.2/configuration/authentication/#transport-strategy-authentication-backend) so the `current_active_user` can recognize and use your custom token.

You can take inspiration from existing ones: https://github.com/fastapi-users/fastapi-users/blob/master/fastapi_users/authentication/strategy/redis.py
